src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/subquery/model/SubqueryType.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.parser.subquery.model;
+
+public enum SubqueryType {
+    /**
+     * SELECT * FROM A WHERE a IN (SELECT * FROM B)
+     */
+    IN(0),
+    UNSUPPORTED(32),
+    NOT_SUBQUERY(33);
```

Comment:
I second comment that Arsen has made earlier. That 
NOT_SUBQUERY feels very unnatural. I think that it shouldn't be a part of this enum. The check if query is not supported should be made earlier (e.g. SubqueryRewriterHelper) / elsewhere (e.g. extract that logic to separate class), and reuse it.

In this case everwhere code will get Subquery, it will know that this is a subquery, without additional checks.

In it's current form it reminds me a meme

enum Boolean {
    TRUE,
    FALSE,
    FILE_NOT_FOUND
}

:)

Reply:
Done.